### PR TITLE
feat(object): add tags to buckets

### LINKED ISSUE #499: Add tags management for scaleway_object_bucket
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add tags management for `scaleway_object_bucket`  resource.

### New or Affected Resource(s)

* scaleway_object_bucket

### Potential Terraform Configuration

Example

```hcl
...
tags = {
    Foo = "Bar"
    Bar = "Baz"
}
...
```

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

Fixes #499

### CODE REVIEW

File: scaleway/resource_object_bucket.go [hunk 1]
```diff
@@ -94,6 +124,35 @@ func resourceScalewayObjectBucketRead(d *schema.ResourceData, m interface{}) err
...
+		if tagSet.Key != nil {
+			key = *tagSet.Key
+		}
+		if tagSet.Value != nil {
+			value = *tagSet.Value
+		}
+		tags[key] = value
+	}
+
+	_ = d.Set("tags", tags)
```
Comment (jerome-quere) [inline scaleway/resource_object_bucket.go:154]: Could you follow the `expand`/ `flatten` pattern for this port of the code

File: scaleway/resource_object_bucket.go [hunk 2]
```diff
@@ -116,6 +175,24 @@ func resourceScalewayObjectBucketUpdate(d *schema.ResourceData, m interface{}) e
...
 
+	if d.HasChange("tags") {
+		tagsSet := make([]*s3.Tag, 0)
+
+		for key, value := range d.Get("tags").(map[string]interface{}) {
+			tagsSet = append(tagsSet, &s3.Tag{
+				Key:   &key,
+				Value: scw.StringPtr(value.(string)),
+			})
+		}
```
Comment (jerome-quere) [inline scaleway/resource_object_bucket.go:186]: Could you follow the expand/ flatten pattern for this port of the code

File: scaleway/resource_object_bucket.go [hunk 3]
```diff
@@ -60,6 +69,27 @@ func resourceScalewayObjectBucketCreate(d *schema.ResourceData, m interface{}) e
...
 	}
 
+	tagsSet := make([]*s3.Tag, 0)
+
+	for key, value := range d.Get("tags").(map[string]interface{}) {
+		tagsSet = append(tagsSet, &s3.Tag{
+			Key:   &key,
+			Value: scw.StringPtr(value.(string)),
+		})
+	}
```
Comment (jerome-quere) [inline scaleway/resource_object_bucket.go:79]: Could you follow the expand/ flatten pattern for this port of the code

File: website/docs/r/object_bucket.html.markdown [hunk 4]
```diff
@@ -23,6 +26,7 @@ resource "scaleway_object_bucket" "some_bucket" {
 The following arguments are supported:
 
 * `name` - (Required) The name of the bucket.
+* `tags` - (Optional) The tags of the bucket.
```
Comment (jerome-quere) [inline website/docs/r/object_bucket.html.markdown:29]:
```suggestion
* `tags` - (Optional) A list of tags (key / value) for the bucket. 
```

**Review by remyleone — APPROVED** [review]

**Review by jerome-quere — APPROVED** [review]

**Pull request merged**